phpunit/class-dynamic-blocks-render-test.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Dynamic blocks rendering Test
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Test do_blocks
+ */
+class Dynamic_Blocks_Render_Test extends WP_UnitTestCase {
+	/**
+	 * Dummy block rendering function.
+	 *
+	 * @param  array $attributes Block attributes.
+	 *
+	 * @return string             Block output.
+	 */
+	function render_dummy_block( $attributes ) {
+		return $attributes['value'];
+	}
+
+	/**
+	 * Successfull dynamic block rendering
+	 */
+	function test_register_block() {
```

Comment:
In this PR there have been issues with matching the content inside the block delimiters. Let's get another test similar to this one but testing out content inside blocks as well:

```html
before
<!-- wp:core/dummy value:a1 -->this should be ignored<!-- /wp:core/dummy -->
between
<!-- wp:core/dummy value:a2 -->this should be ignored also<!-- /wp:core/dummy -->
after
```